docs/concepts/what-is-an-activity-definition.md

```diff
@@ -7,10 +7,35 @@ tags:
   - explanation
 ---
 
-An Activity Definition is the code that defines the constraints of an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution).
+An Activity Definition is the set of constraints that apply to an [Activity Execution](/docs/concepts/what-is-an-activity-execution). Activity Definitions are referred to by their [Activity Type](/docs/concepts/what-is-an-activity-type).
 
-We strongly recommend that you develop an Activity Definition in a language that has a corresponding Temporal SDK.
+In Temporal documentation, the term 'Activity Definition' will be used to refer to an Activity Function Definition——the method or function that is invoked for an [Activity Task Execution](/docs/concepts/what-is-an-activity-task-execution). Some SDKs refer to Activity Definitions as Activity Functions, Activity Interfaces, or Activity Methods. All of these terms refer to the source of an instance of an execution, with an Activity Interface being a collective group of functions.
+
+![Activity Definition](/static/diagrams/activity-definition.svg)
+
+## Constraints
+
+Temporal does not recover Activity state in the event of failure. Therefore, an Activity Definition has no restrictions on the code it contains.
```

Comment:
I would double down on this message to make it clear.

Activity functions are executed as normal functions.
In the event of failure, if the function is executed again, it begins at initial state (with the exception of when Activity Heartbeats are involved).
Therefore - there is no restrictions on the code -